Tests/JobsTests/Scheduler/RecurrenceRuleConstraintTests.swift

```diff
@@ -0,0 +1,195 @@
+    }
+
+    func testRecurrenceRuleConstraintCreationStep() throws {
+        // second (0-59)
+        XCTAssertThrowsError(try SecondRecurrenceRuleConstraint.secondStep(0))
+        XCTAssertNoThrow(try SecondRecurrenceRuleConstraint.secondStep(1))
+        XCTAssertNoThrow(try SecondRecurrenceRuleConstraint.secondStep(59))
+        XCTAssertThrowsError(try SecondRecurrenceRuleConstraint.secondStep(60))
+
+        // minute (0-59)
+        XCTAssertThrowsError(try MinuteRecurrenceRuleConstraint.minuteStep(0))
+        XCTAssertNoThrow(try MinuteRecurrenceRuleConstraint.minuteStep(1))
+        XCTAssertNoThrow(try MinuteRecurrenceRuleConstraint.minuteStep(59))
+        XCTAssertThrowsError(try MinuteRecurrenceRuleConstraint.minuteStep(60))
+
+        // hour (0-23)
+        XCTAssertThrowsError(try HourRecurrenceRuleConstraint.hourStep(0))
+        XCTAssertNoThrow(try HourRecurrenceRuleConstraint.hourStep(1))
+        XCTAssertNoThrow(try HourRecurrenceRuleConstraint.hourStep(23))
+        XCTAssertThrowsError(try HourRecurrenceRuleConstraint.hourStep(24))
+
+        // dayOfWeek (1-7) ex: 1 sunday, 7 saturday
+        XCTAssertThrowsError(try DayOfWeekRecurrenceRuleConstraint.dayOfWeekStep(0))
+        XCTAssertNoThrow(try DayOfWeekRecurrenceRuleConstraint.dayOfWeekStep(1))
+        XCTAssertNoThrow(try DayOfWeekRecurrenceRuleConstraint.dayOfWeekStep(7))
+        XCTAssertThrowsError(try DayOfWeekRecurrenceRuleConstraint.dayOfWeekStep(8))
+
+        // dayOfMonth (1-31) ex: 1 is the 1st of month, 31 is the 31st of month
+        XCTAssertThrowsError(try DayOfMonthRecurrenceRuleConstraint.dayOfMonthStep(0))
+        XCTAssertNoThrow(try DayOfMonthRecurrenceRuleConstraint.dayOfMonthStep(1))
+        XCTAssertNoThrow(try DayOfMonthRecurrenceRuleConstraint.dayOfMonthStep(31))
+        XCTAssertThrowsError(try DayOfMonthRecurrenceRuleConstraint.dayOfMonthStep(32))
+
+        // month (1-12) ex: 1 is January, 12 is December
+        XCTAssertThrowsError(try MonthRecurrenceRuleConstraint.monthStep(0))
+        XCTAssertNoThrow(try MonthRecurrenceRuleConstraint.monthStep(1))
+        XCTAssertNoThrow(try MonthRecurrenceRuleConstraint.monthStep(12))
+        XCTAssertThrowsError(try MonthRecurrenceRuleConstraint.monthStep(53))
+
+        // quarter (1-4)
+        XCTAssertThrowsError(try QuarterRecurrenceRuleConstraint.quarterStep(0))
+        XCTAssertNoThrow(try QuarterRecurrenceRuleConstraint.quarterStep(1))
+        XCTAssertNoThrow(try QuarterRecurrenceRuleConstraint.quarterStep(4))
+        XCTAssertThrowsError(try QuarterRecurrenceRuleConstraint.quarterStep(5))
+
+        // year (1970-3000)
+        XCTAssertThrowsError(try YearRecurrenceRuleConstraint.yearStep(0))
+        XCTAssertNoThrow(try YearRecurrenceRuleConstraint.yearStep(2))
+        XCTAssertNoThrow(try YearRecurrenceRuleConstraint.yearStep(1000))
+        //        XCTAssertThrowsError(try reccurrenceRule.atYear(3001))
+    }
+
+    static var allTests = [
```

Comment:
We shouldn't need `allTests` in any of our `XCTestCase` sub-classes as long as we run `swift test --generate-linuxmain` before committing.